### ISSUE TITLE
Rename `AllocatorSelector::BumpPointer` to `AllocatorSelector::Bump`?

### ISSUE THREAD

**Issue by k-sareen** [mmtk/mmtk-core#1002]
We now have two enums who have a variant named `BumpPointer`: [`AllocatorSelector`](https://github.com/mmtk/mmtk-core/blob/a58246d6bf627423b2e32c78d193bed57c1f5202/src/util/alloc/allocators.rs#L185) and [`AllocatorInfo`](https://github.com/mmtk/mmtk-core/blob/a58246d6bf627423b2e32c78d193bed57c1f5202/src/util/alloc/allocators.rs#L200-L202). Perhaps we should rename `AllocatorSelector::BumpPointer` to `AllocatorSelector::Bump` or something else to avoid confusion.

**Comment by qinsoon** [comment]
It is fine to have variants with the same name in different enums. `AllocatorSelector::BumpPointer` is different from `AllocatorInfo::BumpPointer`. 

However, there is some confusion here. `AllocatorSelector::BumpPointer` refers to the `BumpAllocator`, while `AllocatorInfo::BumpPointer` refers to `BumpPointer`. The root cause is `BumpAllocator` is confusing with `BumpPointer`. We should sort that out, and if necessary, change the variant name in those enums.

**Comment by wks** [comment]
Agreed.  `AllocatorSelector` selects allocator, and `AllocatorInfo` returns offsets of fast-path structures, such as `BumpPointer`.  I think `AllocatorInfo::BumpPointer` is OK, but `AllocatorSelector::BumpPointer` is not a proper name because both `BumpAllocator` and `ImmixAllocator` use `BumpPointer` structs, making it confusing.

I think we can make it consistent by making the variants of `AllocatorSelector` the name of the allocator type without the "-Allocator" suffix, such as `FreeListAllocator` -> `FreeList`, `BumpAllocator` -> `Bump`, etc.

**Comment by k-sareen** [comment]
Actually -- there are three `BumpPointer`s. There's the [actual `BumpPointer` struct](https://github.com/mmtk/mmtk-core/blob/d253d97d96766b31fd0014b6c98ec34728cb62ee/src/util/alloc/bumpallocator.rs#L34-L37) which represents the cursor and limit. I just got very confused reading the ["how to implement fastpath"](https://docs.mmtk.io/portingguide/perf_tuning/alloc.html#option-3-embed-the-fastpath-struct) documentation. I thought the `BumpPointer` in the documentation was referring to `AllocatorInfo::BumpPointer` which does not have the cursor and limit but just the offset. I think this should really be fixed.
